app/assets/javascripts/govuk_publishing_components/analytics-ga4/gtm-click-tracking.js

```diff
@@ -25,11 +25,25 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
           link_url: window.location.href.substring(window.location.origin.length),
           ui: JSON.parse(target.getAttribute('data-gtm-attributes')) || {}
         }
-        var ariaExpanded = this.checkExpandedState(target)
-        if (ariaExpanded) {
-          data.ui.state = ariaExpanded === 'false' ? 'opened' : 'closed'
+        var ariaExpanded = this.getClosestAttribute(target, 'aria-expanded')
+        /*
+          the details component uses an 'open' attribute instead of aria-expanded, so we need to check if we're on a details component.
+          since details deletes the 'open' attribute when closed, we need this boolean, otherwise every element which
+          doesn't contain an 'open' attr would be pushed to gtm as a closed element.
+        */
+        var canHaveOpenAttribute = target.getAttribute('data-module') ? target.getAttribute('data-module').includes('govuk-details') : false
```

Comment:
I'm interested by your approach - why did you choose to look for the specific `data-module` rather than the `details` element itself?

Reply:
This was me being overly cautious - I was not sure if that would detect `<details>` elements that aren't part of the GOV.UK components. But thinking about it now, we're already doing a blind check for just `aria-expanded` anyway so that isn't an issue. This is just me forgetting how the GOV.UK ecosystem works! I'll change it to check the element itself. Thanks @andysellick :+1: